src/api/cvc4cpp.h

```diff
@@ -0,0 +1,2372 @@
+  friend struct SortHashFunction;
+  friend class Term;
+
+  public:
+    /**
+     * Copy constructor.
+     */
+    Sort(const Sort& s);
+
+    /**
+     * Destructor.
+     */
+    ~Sort();
+
+    /**
+     * Assignment operator.
+     * @param s the sort to assign to this sort
+     * @return this sort after assignment
+     */
+    Sort& operator=(const Sort& s);
+
+    /**
+     * Comparison for structural equality.
+     * @param s the sort to compare to
+     * @return true if the sorts are equal
+     */
+    bool operator==(const Sort& s) const;
+
+    /**
+     * Comparison for structural disequality.
+     * @param s the sort to compare to
+     * @return true if the sorts are not equal
+     */
+    bool operator!=(const Sort& s) const;
+
+    /**
+     * Is this a Boolean sort?
+     * @return true if the sort is a Boolean sort
+     */
+    bool isBoolean() const;
+
+    /**
+     * Is this a integer sort?
+     * @return true if the sort is a integer sort
+     */
+    bool isInteger() const;
+
+    /**
+     * Is this a real sort?
+     * @return true if the sort is a real sort
+     */
+    bool isReal() const;
+
+    /**
+     * Is this a string sort?
+     * @return true if the sort is the string sort
+     */
+    bool isString() const;
+
+    /**
+     * Is this a regexp sort?
+     * @return true if the sort is the regexp sort
+     */
+    bool isRegExp() const;
+
+    /**
+     * Is this a rounding mode sort?
+     * @return true if the sort is the rounding mode sort
+     */
+    bool isRoundingMode() const;
+
+    /**
+     * Is this a bit-vector sort?
+     * @return true if the sort is a bit-vector sort
+     */
+    bool isBitVector() const;
+
+    /**
+     * Is this a floating-point sort?
+     * @return true if the sort is a floating-point sort
+     */
+    bool isFloatingPoint() const;
+
+    /**
+     * Is this a datatype sort?
+     * @return true if the sort is a datatype sort
+     */
+    bool isDatatype() const;
+
+    /**
+     * Is this a function sort?
+     * @return true if the sort is a function sort
+     */
+    bool isFunction() const;
+
+    /**
+     * Is this a predicate sort?
+     * That is, is this a function sort mapping to Boolean? All predicate
+     * sorts are also function sorts.
+     * @return true if the sort is a predicate sort
+     */
+    bool isPredicate() const;
+
+    /**
+     * Is this a tuple sort?
+     * @return true if the sort is a tuple sort
+     */
+    bool isTuple() const;
+
+    /**
+     * Is this a record sort?
+     * @return true if the sort is a record sort
+     */
+    bool isRecord() const;
+
+    /**
+     * Is this an array sort?
+     * @return true if the sort is a array sort
+     */
+    bool isArray() const;
+
+    /**
+     * Is this a Set sort?
+     * @return true if the sort is a Set sort
+     */
+    bool isSet() const;
+
+    /**
+     * Is this a sort kind?
+     * @return true if this is a sort kind
+     */
+    bool isSort() const;
```

Comment:
This is a bit confusing since the class is called `Sort`. Is there any other name that we could use?